pkg/function/checkRepository.go

```diff
@@ -0,0 +1,126 @@
+)
+
+const (
+	// CheckRepositoryArtifactPrefixArg provides the path to restore backed up data
+	CheckRepositoryArtifactPrefixArg = "backupArtifactPrefix"
+	// CheckRepositoryEncryptionKeyArg provides the encryption key to be used for deletes
+	CheckRepositoryEncryptionKeyArg = "encryptionKey"
+	// CheckRepositoryPodOverrideArg contains pod specs to override default pod specs
+	CheckRepositoryPodOverrideArg    = "podOverride"
+	CheckRepositoryJobPrefix         = "check-repository-"
+	CheckRepositoryPasswordIncorrect = "passwordIncorrect"
+	CheckRepositoryRepoDoesNotExist  = "repoUnavailable"
+)
+
+func init() {
+	kanister.Register(&CheckRepositoryFunc{})
+}
+
+var _ kanister.Func = (*CheckRepositoryFunc)(nil)
+
+type CheckRepositoryFunc struct{}
+
+func (*CheckRepositoryFunc) Name() string {
+	return "CheckRepository"
+}
+
+func CheckRepository(ctx context.Context, cli kubernetes.Interface, tp param.TemplateParams, encryptionKey, targetPaths, jobPrefix string, podOverride crv1alpha1.JSONMap) (map[string]interface{}, error) {
+	namespace, err := kube.GetControllerNamespace()
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to get controller namespace")
+	}
+	options := &kube.PodOptions{
+		Namespace:    namespace,
+		GenerateName: jobPrefix,
+		Image:        kanisterToolsImage,
+		Command:      []string{"sh", "-c", "tail -f /dev/null"},
+		PodOverride:  podOverride,
+	}
+	pr := kube.NewPodRunner(cli, options)
+	podFunc := CheckRepositoryPodFunc(cli, tp, namespace, encryptionKey, targetPaths)
+	return pr.Run(ctx, podFunc)
+}
+
+func CheckRepositoryPodFunc(cli kubernetes.Interface, tp param.TemplateParams, namespace, encryptionKey, targetPath string) func(ctx context.Context, pod *v1.Pod) (map[string]interface{}, error) {
+	return func(ctx context.Context, pod *v1.Pod) (map[string]interface{}, error) {
+		// Wait for pod to reach running state
+		if err := kube.WaitForPodReady(ctx, cli, pod.Namespace, pod.Name); err != nil {
+			return nil, errors.Wrapf(err, "Failed while waiting for Pod %s to be ready", pod.Name)
+		}
+		pw, err := GetPodWriter(cli, ctx, pod.Namespace, pod.Name, pod.Spec.Containers[0].Name, tp.Profile)
+		if err != nil {
+			return nil, err
+		}
+		defer CleanUpCredsFile(ctx, pw, pod.Namespace, pod.Name, pod.Spec.Containers[0].Name)
+		err = restic.CheckIfRepoIsReachable(tp.Profile, targetPath, encryptionKey, cli, namespace, pod.Name, pod.Spec.Containers[0].Name)
+		switch {
+		case err == nil:
+			break
+		case strings.Contains(err.Error(), restic.PasswordIncorrect):
+			return map[string]interface{}{
+					CheckRepositoryPasswordIncorrect: "true",
+					CheckRepositoryRepoDoesNotExist:  "false",
+				},
+				nil
+
+		case strings.Contains(err.Error(), restic.RepoDoesNotExist):
+			return map[string]interface{}{
+
+					CheckRepositoryPasswordIncorrect: "false",
+					CheckRepositoryRepoDoesNotExist:  "true",
+				},
+				nil
+		default:
+			return nil, err
+
+		}
+		return map[string]interface{}{
```

Comment:
[minor] this code is unreachable

Reply:
We break out of `switch` is `err==nil`;  https://github.com/kanisterio/kanister/pull/382/files#diff-7ef89596544af2a5c43f68980656f28aR72

This will be returned then

Reply:
:+1:  I missed the break